### PR TITLE
Clean up TODO and XXX comments

### PR DESCRIPTION
**Pull request by Kodiologist**

### CODE REVIEW

File: docs/language/internals.rst [hunk 1]
```diff
@@ -436,10 +436,3 @@ Our final version of ``nif``, built with ``defmacro/g!`` becomes::
 
 Checking Macro Arguments and Raising Exceptions
 -----------------------------------------------
-
-
-
-Hy Compiler Built-Ins
-=====================
-
-.. TODO: Write this.
```
Comment (peaceamongworlds) [inline docs/language/internals.rst:445]: This leaves an empty subheading in the docs. It should also probably get cleaned up.

Comment (Kodiologist) [reply docs/language/internals.rst]: Good point. Done.

**Review by peaceamongworlds — APPROVED** [review]

**Comment by Kodiologist** [comment]
@hylang/core This PR adds some tips on merging to CONTRIBUTING that can be a good reference.

File: hy/contrib/walk.hy [hunk 2]
```diff
@@ -431,7 +428,7 @@
         (= head 'defclass) (self.handle-defclass)
         (= head 'quasiquote) (self.+quote)
         ;; must be checked last!
-        (in (mangle head) special-forms) (self.handle-special-form)
+        (in head (special)) (self.handle-special-form)
```
Comment (scauligi) [inline hy/contrib/walk.hy:431]: Comparing unmangled forms can miss (obtuse) cases like `(eval_when_compile ...)`

File: hy/extra/reserved.hy [hunk 3]
```diff
@@ -7,15 +7,20 @@
 
 (setv _cache None)
 
+(defn special []
+  "Return a frozenset of special operators, such as ``defn`` and ``+``."
```
Comment (scauligi) [inline hy/extra/reserved.hy:11]: `defn` is actually a macro

**Comment by Kodiologist** [comment]
@scauligi I fixed those. Thanks.

**Review by scauligi — APPROVED** [review]

**Pull request merged**